### ISSUE TITLE
Cross-browser plain-text copy/paste does not seem to work

### ISSUE THREAD

**Issue by rwaters71** [froala/wysiwyg-editor#513]
When using Froala under Chrome, for example, if you paste any text copied from a different browser (IE11 in my case), Froala adds table wrapper around the pasted text.  This behavior occurs regardless of the fact the editor is invoked with ```plainPaste: true```

Additional notes: Opted for using Chrome, because Froala becomes very sluggish under IE when editing larger text areas, especially after pasting any text in it. Above experienced using Froala 1.27

**Comment by stefanneculai** [comment]
Please send us a link to what you're copying.

**Comment by rwaters71** [comment]
Ah, you are right that it does not do it with all content. Any paste from vBulletin posts (any paragraph) would do it:  http://hardforum.com/showthread.php?t=1858181
Wraps it in a table and adds comments inside the td element: ```<!--StartFragment-->``` ... ```<!--EndFragment-->```

Seems that it depends on the source site. Here is another example:  http://www.raspberrypi.org/opensprinkler-pi/
Copying part of a sentence on this site (using IE) and pasting in Chrome (with supposedly plaintext), pastes all this:
`````
<title>OpenSprinkler Pi | Raspberry Pi</title>

<p>                    </p>
<!--[if lt IE 9]>        <![endif]-->
<p>                                                                                                                                                                </p>
<!-- This site uses the Google Analytics by Yoast plugin v5.3.3 - Universal enabled - https://yoast.com/wordpress/plugins/google-analytics/ -->
<p>  </p>
<!-- / Google Analytics by Yoast -->
<p>             </p>
<section><article>
<p><!--StartFragment-->tools from the community like Rich Zimmerman’s <!--EndFragment--></p>
</article></section>
<p><br>
</p>
`````

**Comment by kotmatpockuh** [comment]
same issue to me...

**Comment by hws70696** [comment]
same bug
@stefanneculai  when you ~ will fix it?

**Comment by stefanneculai** [comment]
Thank you all for bringing this to our attention. We'll fix this as soon as we can.

**Comment by kotmatpockuh** [comment]
my temporary hotfix in angular:
```
onPaste: function(e, editor, html) {
            if (!angular.isUndefined(html)){
              $scope.parser = new DOMParser();
              $scope.doc = $scope.parser.parseFromString(html, "text/html");
              $scope.cpText = $scope.doc.getElementsByClassName('froala-view froala-element f-basic');
              if (!angular.isUndefined($scope.cpText) && $scope.cpText.length > 0){
                return $scope.cpText[0].innerHTML;
              }
            }
          }
```

**Comment by qrz777** [comment]
hey)
same issue
fix it ASAP please

**Comment by kotmatpockuh** [comment]
do you plan to fix it on this week?

**Comment by stefanneculai** [comment]
We will fix it as soon as we can, but I can't tell you a specific date.

**Comment by ojs41564** [comment]
19 days!
and this plugin is not free...

from expensive plugin everybody want's fast bug support.

**Comment by stefanneculai** [comment]
Bugs reported by customers with a license have higher priority. We haven't had any email on https//froala.com/contact about this issue so far and that's why it wasn't fixed yet.

**Comment by kotmatpockuh** [comment]
it's still reproducable
whole html appears when copying from ie

**Comment by stefanneculai** [comment]
It doesn't in plainText mode. When plainText is set to false the tags that are kept are the one specified in allowedTags option.

**Comment by sadiashujaat93-dev** [comment]
I faced a similar issue recently where cross-browser plain-text copy/paste behaved differently in Chrome vs Firefox, especially when copying from rich-text editors. In my case, the problem was caused by hidden formatting and inconsistent handling of line breaks (\n vs \r\n). One solution that helped was sanitizing the clipboard input and forcing plain text paste using JavaScript clipboard events.

I also tested the output using my <a href=https://syntaxai.es/>syntax analyzer</a> tool to confirm whether extra invisible characters were being added during paste. It made debugging much easier because I could instantly see what characters were being inserted.

Has anyone else noticed this issue specifically when copying from Word or Google Docs?
